sdk-cpp/src/internal/do_config_internal.h

```diff
@@ -0,0 +1,12 @@
+#ifndef _DELIVERY_OPTIMIZATION_DO_CONFIG_INTERNAL_H
```

Comment:
What is the need to create this internal config API? Is it because it cannot be converted to not use try/catch?

Reply:
The public header needs to be platform agnostic, particularly for cases where we build & install the sdk as a package. Sorry - this PR is confusing because I added whats really two changes into one PR.

Reply:
Are you going to separate it as discussed? To release a new version for DU...

Reply:
Yes I can look into splitting this PR into two. I'll resolve all current comments, and re-open two separate PR's, the first change will be the platform agnostic headers (which we can set as the release commit if things look good), and then the exceptions change.